### DISCUSSION TITLE
Additional random numbers inside mi.BSDF.sample

### DISCUSSION THREAD

**Discussion by SimonLCS** [mitsuba-renderer/mitsuba3#1053] [Q&A]
Hello,

I trying to implement a multiple scattering BSDF ([Heitz2016](https://eheitzresearch.wordpress.com/240-2/)) inside a python BSDF plugin but I'm facing a problem.
I need to sample multiple directions inside the mi.BSDF.sample function but I only have access to 3 random values (sample1 : mi.Float, sample2 : mi.Point2), which is, to the best of my knowledge insufficient.
Is there a way to generate additional random numbers inside the mi.BSDF.sample function ?

Simon LUCAS,
Thanks in advance !

**Comment by njroussel** [comment]
Hi @SimonLCS 

There is a  simple solution, but it assumes you want to sample the BSDF independently from your main path tracer sampler:
* You build a new sampler in the `sample` function. You can initialize its seed with one of the random samples you can as input.

There are some Dr.Jit-related technical details at play here that make it difficult to pass an existing sampler to the metod call. (In addition to it requiring changing interfaces).

By the way, what I mentioned above is only really relevant in JIT-enabled variants. For a scalar variant, you're better off just stroing a separate sampler in your BSDF and using that in your sampling routines.
